lib/ui/painting/vertices.cc

```diff
@@ -67,7 +67,7 @@ bool Vertices::init(Dart_Handle vertices_handle,
     builder.store_colors(reinterpret_cast<const SkColor*>(colors.data()));
   }
 
-  if (indices.data()) {
+  if (indices.data() && indices.num_elements() > 0) {
```

Comment:
We talked this out offline, but the data conditions here are:

pointers, texCoords, colors are all the same size (well, 2 of them are x,y so they are twice as long, but basically their counts are tied together). This is because all of that data is tied together 1:1:1.

Indices has its own count because it can have a different size than the others.

The Builder constructor (for both SkVertices and DlVertices) takes flags to indicate if the arrays are being delivered, but the indices passes in the count as an indicator (I could have switched this to a boolean, but kept the same conventions as SkVertices).

So, pointers is copied unconditionally since it is required, even if it is zero length.
texCoords and colors are copied on the same condition as their flags were set - by nullness.
indices is copied on the same condition as its indicator in the builder constructor (non-null and count > 0).

I just noticed a simplification. elements() > 0 covers the null case as well. But I'm not going to redo the PR as it needs to propagate. The optimization is minimal.